src/android/com/adobe/phonegap/push/FCMService.java

```diff
@@ -329,6 +330,12 @@ private void showNotificationIfPossible(Context context, Bundle extras) {
     Log.d(LOG_TAG, "title =[" + title + "]");
     Log.d(LOG_TAG, "contentAvailable =[" + contentAvailable + "]");
     Log.d(LOG_TAG, "forceStart =[" + forceStart + "]");
+    Log.d(LOG_TAG, "clearNotifications =[" + clearNotifications + "]");
+
+    if ("true".equals(clearNotifications)) {
```

Comment:
Users may send clearNotifications as a string or as a boolean so we should expect them to mess up and check string equality for "true" and if it is the boolean `true`.

Reply:
I was unable to retrieve the value as a boolean even when sent with `"clearAllNotifications": true`. Java claimed it was a string every time, so this is why I set it to a string.

Any ideas why?

Reply:
Not without looking into it in more detail but I'd bet it is some kinda of screwy JSON serialization  issue on Android.

Reply:
I've made some of the changes but I'll leave this one for later.